src/modm/ui/menu/choice_menu.hpp

```diff
@@ -112,4 +114,6 @@ namespace modm{
 	};
 }
 
+#include "choice_menu_impl.hpp"
```

Comment:
Why is this being included at the end of the file?

Reply:
Doing this is a way of creating a template class with separate files for declaration/definition. Since you can't put template stuff in a `.cpp` file you put it in a `_impl.hpp` file and then include it at the bottom of the normal `.hpp` file. Then you can add a preprocessor statement at the top of the `_impl.hpp` file that makes it so your code won't compile if you try and include the `_impl.hpp` file in your own code (since you should be including the `.hpp` file instead).